### PR TITLE
Add support for Scala 2.12

### PR DESCRIPTION
**Pull request by Thangiee**
Twitter uses this library and would appreciate updating it to Scala 2.12. :)

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+val scala211 = "2.11.8"
+
+lazy val settings =
+  Seq(
+    organization := "org.msgpack",
+    name := "msgpack-scala",
+    version := messagePackVersion,
+    scalaVersion := scala211,
+    crossScalaVersions := Seq("2.10.5", scala211, "2.12.1"), // After 2.10 ,binaries are compatible.So don't need to crossCompile.(tests are passed even comment outed versions.)
```
Comment (logarithm) [inline build.sbt]: I would suggest using JDK 1.7 for scala versions oldest than 2.12.

File: build.sbt [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    resolvers ++= Seq(Resolver.mavenLocal),
+    parallelExecution in Test := false,
+    scalacOptions += {
+      scalaVersion.value match {
+        case v if v.startsWith("2.12") => "-target:jvm-1.8"
```
Comment (logarithm) [inline build.sbt]: Should it be `Seq("-target:jvm-1.8")` ?

**Comment by Thangiee** [comment]
Any chance of getting this merge?

**Comment by holinov** [comment]
+1. also whant this MR

**Comment by sammerry** [comment]
+1

**Comment by xerial** [comment]
Hi. We no longer maintain msgpack-scala 0.6.x, which is based on a deprecated version of msgpack-java 0.6.x. In #25, we will start a new version development.

msgpack-scala 0.8.x will add new feature set specific to Scala, such as case class to msgpack conversion, etc.

**Comment by xerial** [comment]
Thanks for you making this PR, but let me close this in favor of #25.

**Pull request closed without merging**